### PR TITLE
Remove node exporter k8scc

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#1902

Updates v12 to use k8scc 3.3.2 which has node-exporter resources removed. So it can be managed by chart-operator.

### CODE REVIEW

File: service/controller/v12/adapter/launch_configuration_test.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func TestAdapterLaunchConfigurationSmallCloudConfig(t *testing.T) {
 		},
 		{
 			description:  "s3 http uri",
-			expectedLine: `s3_http_uri="https://s3.myregion.amazonaws.com/000000000000-g8s-test-cluster/cloudconfig/v_3_3_1/$USERDATA_FILE"`,
+			expectedLine: `s3_http_uri="https://s3.myregion.amazonaws.com/000000000000-g8s-test-cluster/cloudconfig/v_3_3_2/$USERDATA_FILE"`,
```
Comment (xh3b4sd) [inline service/controller/v12/adapter/launch_configuration_test.go]: Is this test actually useful? I am thinking to remove this.

File: service/controller/v12/cloudconfig/master_template.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ import (
 const (
 	// MasterCloudConfigVersion defines the version of k8scloudconfig in use.
 	// It is used in the main stack output and S3 object paths.
-	MasterCloudConfigVersion = "v_3_3_1"
+	MasterCloudConfigVersion = "v_3_3_2"
```
Comment (xh3b4sd) [inline service/controller/v12/cloudconfig/master_template.go]: We could combine the master and worker constants to only have one which we have to update.

File: service/controller/v12/resource/s3object/desired_test.go [hunk 3]
```diff
@@ -24,21 +26,23 @@ func Test_DesiredState(t *testing.T) {
 		},
 	}
 
+	masterKeyPattern := "cloudconfig/v[\\d_]+/master"
+	workerKeyPattern := "cloudconfig/v[\\d_]+/worker"
```
Comment (rossf7) [inline service/controller/v12/resource/s3object/desired_test.go]: I refactored the test to match on a regex so we only need to update the constant in the cloudconfig service.

File: service/controller/v12/adapter/launch_configuration_test.go (hunk 1 above)
Comment (rossf7) [reply service/controller/v12/adapter/launch_configuration_test.go]: I removed this test case but see below I refactored the s3object test to use a regex. So I don't think we lose much by removing this case.

File: service/controller/v12/cloudconfig/master_template.go (hunk 2 above)
Comment (rossf7) [reply service/controller/v12/cloudconfig/master_template.go]: Good idea. I combined the 2 constants. For future updates we just vendor k8scc and update the constant.

File: service/controller/v12/cloudconfig/spec.go [hunk 4]
```diff
@@ -4,6 +4,12 @@ import (
 	"github.com/aws/aws-sdk-go/service/kms"
 )
 
+const (
+	// CloudConfigVersion defines the version of k8scloudconfig in use.
+	// It is used in the main stack output and S3 object paths.
+	CloudConfigVersion = "v_3_3_2"
```
Comment (rossf7) [inline service/controller/v12/cloudconfig/spec.go]: This is the new constant.

Comment (kopiczko) [reply service/controller/v12/cloudconfig/spec.go]: May be an idea to put it in k8scloudconfig repo.

Comment (rossf7) [reply service/controller/v12/cloudconfig/spec.go]: Good idea but going with this. It improves the situation as there is now only a single constant and no changes to tests.

We need to make more k8scc changes for managed services so we can come back to this.

**Comment by rossf7** [comment]
CI green. Going with this.

**Pull request merged**